mbtiles/src/metadata.rs

```diff
@@ -231,33 +232,43 @@ impl Mbtiles {
     }
 }
 
+pub async fn anonymous_mbtiles(script: &str) -> (Mbtiles, SqliteConnection) {
+    let mbt = Mbtiles::new(":memory:").unwrap();
+    let mut conn = mbt.open().await.unwrap();
+    sqlx::raw_sql(script).execute(&mut conn).await.unwrap();
+    (mbt, conn)
+}
+
+pub async fn temp_names_mbtiles(file_name: &str, script: &str) -> PathBuf {
```

Comment:
this won't work - you also need to return path and sqliteconnection - otherwise the connection will be dropped on return.  Also, you may want to return a `String` rather than `PathBuf` for simplicity